tests/vm_test.go

```diff
@@ -2088,50 +2063,45 @@ func createRunningVM(virtClient kubecli.KubevirtClient, template *v1.VirtualMach
 
 func startVM(virtClient kubecli.KubevirtClient, vm *v1.VirtualMachine) *v1.VirtualMachine {
 	By("Starting the VirtualMachine")
-	err := tests.RetryWithMetadataIfModified(vm.ObjectMeta, func(meta k8smetav1.ObjectMeta) error {
-		vm, err := virtClient.VirtualMachine(meta.Namespace).Get(context.Background(), meta.Name, k8smetav1.GetOptions{})
-		Expect(err).ToNot(HaveOccurred())
-		vm.Spec.Running = nil
-		runStrategyAlways := v1.RunStrategyAlways
-		vm.Spec.RunStrategy = &runStrategyAlways
-		_, err = virtClient.VirtualMachine(meta.Namespace).Update(context.Background(), vm, metav1.UpdateOptions{})
-		return err
-	})
-	Expect(err).ToNot(HaveOccurred())
+	Expect(patchVmRunStrategy(vm, v1.RunStrategyAlways)).To(Succeed())
 
 	By("Waiting for VMI to be running")
 	Eventually(ThisVMIWith(vm.Namespace, vm.Name), 300*time.Second, 1*time.Second).Should(BeRunning())
 
 	By("Waiting for VM to be ready")
 	Eventually(ThisVM(vm), 360*time.Second, 1*time.Second).Should(BeReady())
 
-	vm, err = virtClient.VirtualMachine(vm.Namespace).Get(context.Background(), vm.Name, k8smetav1.GetOptions{})
+	vm, err := virtClient.VirtualMachine(vm.Namespace).Get(context.Background(), vm.Name, k8smetav1.GetOptions{})
 	Expect(err).ToNot(HaveOccurred())
 
 	return vm
 }
 
 func stopVM(virtClient kubecli.KubevirtClient, vm *v1.VirtualMachine) *v1.VirtualMachine {
 	By("Stopping the VirtualMachine")
-	err := tests.RetryWithMetadataIfModified(vm.ObjectMeta, func(meta k8smetav1.ObjectMeta) error {
-		vm, err := virtClient.VirtualMachine(meta.Namespace).Get(context.Background(), meta.Name, k8smetav1.GetOptions{})
-		Expect(err).ToNot(HaveOccurred())
-		vm.Spec.Running = nil
-		runStrategyHalted := v1.RunStrategyHalted
-		vm.Spec.RunStrategy = &runStrategyHalted
-		_, err = virtClient.VirtualMachine(meta.Namespace).Update(context.Background(), vm, metav1.UpdateOptions{})
-		return err
-	})
-	Expect(err).ToNot(HaveOccurred())
+	Expect(patchVmRunStrategy(vm, v1.RunStrategyHalted)).To(Succeed())
 
 	By("Waiting for VMI to not exist")
 	Eventually(ThisVMIWith(vm.Namespace, vm.Name), 300*time.Second, 1*time.Second).ShouldNot(Exist())
 
 	By("Waiting for VM to not be ready")
 	Eventually(ThisVM(vm), 300*time.Second, 1*time.Second).Should(Not(BeReady()))
 
-	vm, err = virtClient.VirtualMachine(vm.Namespace).Get(context.Background(), vm.Name, k8smetav1.GetOptions{})
+	vm, err := virtClient.VirtualMachine(vm.Namespace).Get(context.Background(), vm.Name, k8smetav1.GetOptions{})
 	Expect(err).ToNot(HaveOccurred())
 
 	return vm
 }
+
+func patchVmRunStrategy(vm *v1.VirtualMachine, strategy v1.VirtualMachineRunStrategy) error {
+	patches := patch.New(
+		patch.WithAdd("/spec/running", nil),
+		patch.WithAdd("/spec/runStrategy", string(strategy)),
+	)
+	patchData, err := patches.GeneratePayload()
+	if err != nil {
+		return err
+	}
+	_, err = kubevirt.Client().VirtualMachine(vm.Namespace).Patch(context.Background(), vm.Name, types.JSONPatchType, patchData, metav1.PatchOptions{})
```

Comment:
can we change it to get the context as a parameter? maybe we could use context properly in the future, but for now, any new `context.Background()` makes it harder to fix.

Reply:
@codingben can you point to where we're doing this right in tests?

Since `context.Background()` is not new here, it has only been moved to this helper function, I would tend to prefer seeing it in a follow-up PR.

Reply:
if you want to improve here one small thing, is to separate between the creation of the patch and its apply.

The the usage will be:
```go
patch := createVMRunStrategyPatch(vm, v1.RunStrategyHalted)
vm, err = kubevirt.Client().VirtualMachine(vm.Namespace).Patch(context.Background(), vm.Name, types.JSONPatchType, patchData, metav1.PatchOptions{})
```

Reply:
@nunnatsa I wanted to ask the same question like @dankenigsberg did, I didn't see the usage of "context as a parameter" in the tests. I'd prefer to keep it as it is in this PR.

@EdDev I can't comment in similar thread of your review. Do you mean to have only `createVMRunStrategyPatch` that will return `patchData` or `err` instead of `patchVmRunStrategy` that returns only `err`?

Reply:
Done.